### PR TITLE
Provide sensible roi and ccd to ITC queries

### PR DESCRIPTION
**Pull request by cquiroz**
There is a mismatch between the exposure time shown in the modes table and what we get from the odb.
The reason is that we are not providing roi nor ccd and the odb sets them depending on the mode an image quality

this PR replicates this logic in explore

### CODE REVIEW

File: model/shared/src/main/scala/queries/schemas/itc/syntax.scala [hunk 1]
```diff
@@ -5,31 +5,65 @@ package queries.schemas.itc
...
 
 trait syntax:
 
   extension (row: InstrumentRow)
-    def toItcClientMode: Option[InstrumentMode] = row match {
+    def toItcClientMode(p: SourceProfile, iq: ImageQuality): Option[InstrumentMode] = row match {
       case g: GmosNorthSpectroscopyRow =>
-        val roi = g.modeOverrides.flatMap(_.roi)
-        val ccd = g.modeOverrides.flatMap(_.ccdMode)
+        val isFPU =
```
Comment (rpiaggio) [inline model/shared/src/main/scala/queries/schemas/itc/syntax.scala:33]: Is this the same logic as in the ODB? If so, can we move it to a common lib? core?

**Review by rpiaggio — APPROVED** [review]
LGTM. Not sure if we are duplicating logic. If we are, I suggest we unify.

File: model/shared/src/main/scala/queries/schemas/itc/syntax.scala (hunk 1 above)
Comment (cquiroz) [reply model/shared/src/main/scala/queries/schemas/itc/syntax.scala:33]: Yes I added this as a method to core

Comment (rpiaggio) [reply model/shared/src/main/scala/queries/schemas/itc/syntax.scala:33]: Nice, I was wondering if the could unify the override logic, but I see now that the overrides are defined in explore.

**Pull request merged**